community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/TemporalSchemaIndexProvider.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import java.io.IOException;
+
+import org.neo4j.index.internal.gbptree.RecoveryCleanupWorkCollector;
+import org.neo4j.internal.kernel.api.IndexCapability;
+import org.neo4j.internal.kernel.api.InternalIndexState;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.api.index.IndexAccessor;
+import org.neo4j.kernel.api.index.IndexDirectoryStructure;
+import org.neo4j.kernel.api.index.IndexPopulator;
+import org.neo4j.kernel.api.index.SchemaIndexProvider;
+import org.neo4j.kernel.api.schema.index.IndexDescriptor;
+import org.neo4j.kernel.impl.api.index.sampling.IndexSamplingConfig;
+import org.neo4j.kernel.impl.storemigration.StoreMigrationParticipant;
+
+public class TemporalSchemaIndexProvider extends SchemaIndexProvider
+{
+    public static final String KEY = "temporal";
+    public static final Descriptor TEMPORAL_PROVIDER_DESCRIPTOR = new Descriptor( KEY, "1.0" );
+
+    private final PageCache pageCache;
+    private final FileSystemAbstraction fs;
+    private final Monitor monitor;
+    private final RecoveryCleanupWorkCollector recoveryCleanupWorkCollector;
+    private final boolean readOnly;
+
+    public TemporalSchemaIndexProvider( PageCache pageCache, FileSystemAbstraction fs,
+            IndexDirectoryStructure.Factory directoryStructure, Monitor monitor,
+            RecoveryCleanupWorkCollector recoveryCleanupWorkCollector, boolean readOnly )
+    {
+        super( TEMPORAL_PROVIDER_DESCRIPTOR, 0, directoryStructure );
+        this.pageCache = pageCache;
+        this.fs = fs;
+        this.monitor = monitor;
+        this.recoveryCleanupWorkCollector = recoveryCleanupWorkCollector;
+        this.readOnly = readOnly;
+    }
+
+    @Override
+    public IndexPopulator getPopulator( long indexId, IndexDescriptor descriptor, IndexSamplingConfig samplingConfig )
+    {
+        if ( readOnly )
+        {
+            throw new UnsupportedOperationException( "Can't create populator for read only index" );
+        }
+        TemporalIndexFiles files = new TemporalIndexFiles( directoryStructure(), indexId, descriptor, fs );
```

Comment:
Seems like there could be a single instance of `TemporalIndexFiles` in this provider, doesn't it? Applicable to all methods that instantiate these

Reply:
That's not really possible as we need the `indexId` and `descriptor`.